subsys/bluetooth/mesh/prov_device.c

```diff
@@ -542,10 +542,18 @@ static const struct bt_mesh_prov_role role_device = {
 
 int bt_mesh_prov_enable(bt_mesh_prov_bearer_t bearers)
 {
+	int err;
+
 	if (bt_mesh_is_provisioned()) {
 		return -EALREADY;
 	}
 
+	err = bt_mesh_adv_enable();
+	if (err) {
+		BT_ERR("Failed enabling advertiser");
+		return err;
+	}
```

Comment:
@joerchan I'm not able to create the extended advertiser in the bt_ready callback until I call settings_load(), because of `BT_DEV_READY`. As a workaround, I have put this here to get the advertiser to start after the settings call, but I have to move it somewhere more appropriate before this can be merged. 

The issue comes from `bt_init()`: 
https://github.com/zephyrproject-rtos/zephyr/blob/698b0a652e48f973b6dcf57c23d07bfb1f4f9fae/subsys/bluetooth/host/hci_core.c#L6486-L6495

The mesh sample gets no id address, and always triggers the "No ID address" log above. This makes this function return early, and bt_init_finalize isn't called until the app calls `settings_load()`. Would it be possible to work around this somehow?

Reply:
You will have to have the identity ready when you create the advertiser, since we need the IRK or the identity address. Since we write all this information to the controller for the advertiser when we create it. 

Since you always get the "No ID address" message does that mean that CONFIG_BT_HCI_VS_EXT is not enabled?
That should be able to read the identity from FICR. But we would still have to handle the case where the application disables this option, or different setups where the controller does not have a static identity.